### ISSUE TITLE
Password reset text template for SMPP code isn't non-ascii compatible

### ISSUE THREAD

**Issue by jamalex** [fle-internal/kolibri-instant-schools-plugin#160]
### Context

* Operating system: Ubuntu
* Context: Voda Educ DRC during upgrade to 0.6.3

### Observed behavior

When there was an accent (French) in the password reset text template, sending password resets threw:

```
Traceback (most recent call last):
  File "kolibri/dist/django/core/handlers/base.py", line 149, in get_response
    response = self.process_exception_by_middleware(e, request)
  File "kolibri/dist/django/core/handlers/base.py", line 147, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "kolibri/dist/django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "kolibri/dist/rest_framework/viewsets.py", line 87, in view
    return self.dispatch(request, *args, **kwargs)
  File "kolibri/dist/rest_framework/views.py", line 466, in dispatch
    response = self.handle_exception(exc)
  File "kolibri/dist/rest_framework/views.py", line 463, in dispatch
    response = handler(request, *args, **kwargs)
  File "kolibri/dist/kolibri_instant_schools_plugin/auth/api.py", line 72, in create
    send_password_reset_link(phone, token.token, baseurl)
  File "kolibri/dist/kolibri_instant_schools_plugin/smpp/utils.py", line 32, in send_password_reset_link
    send_message(phone, conf['SMS_MESSAGE_TEMPLATE'].format(url=url))
  File "kolibri/dist/kolibri_instant_schools_plugin/smpp/utils.py", line 44, in send_message
    parts, encoding_flag, msg_type_flag = smpplib2.gsm.make_parts(message)
  File "kolibri/dist/smpplib2/gsm.py", line 39, in make_parts
    text = gsm_encode(text)
  File "kolibri/dist/smpplib2/gsm.py", line 24, in gsm_encode
    idx = gsm.find(c)
  File "/usr/lib/python2.7/encodings/utf_8.py", line 16, in decode
    return codecs.utf_8_decode(input, errors, True)
UnicodeDecodeError: 'utf8' codec can't decode byte 0xc3 in position 0: unexpected end of data
```

**Comment by jamalex** [comment]
Aaaaand it appears that there's no good way to fix it (without replacing `configparser` with an alternative parser, which is probably the longer-term solution if we keep supporting Python 2.7):

https://bugs.python.org/issue11597

tl;dr adding unicode support to `configparser` was closed as WONTFIX since it would constitute a "feature" and hence can't be added as a patch to Python 2.7.x

**Comment by jamalex** [comment]
Aaaaaaand even after fixing that (by trying things out using Python 3 and tweaking some imports), it turns out that `smpplib2` doesn't support unicode either. Even the SMPP spec itself doesn't seem to always support Unicode, by default just Latin-1, but the gsm library used by smpplib2 doesn't like Latin-1 encoded bytestrings.

So... WONTFIX! Hopefully we can find an accent-free way of phrasing things for Portuguese too. :)

**Comment by jeepurs** [comment]
I think going accent-free will be a creative challenge. Thanks for the update @jamalex !
